sark/data.py

```diff
@@ -111,6 +116,30 @@ def collector(ea, fpos, original, patched):
     return patched_bytes
 
 
+def apply_patches(patched_path=None):
+    to_patch = idaapi.get_input_file_path()
+
+    if patched_path:
+        shutil.copyfile(to_patch, patched_path)
+        to_patch = patched_path
+
+    patches = get_patched_bytes()
+
+    with open(to_patch, "r+b") as output:
+
+        for patch in patches.itervalues():
+
+            output.seek(patch.fpos)
+            curr_byte = ord(output.read(1))
+            if curr_byte != patch.original:
```

Comment:
It's a decision to make.
Relaxing this validation will make our life easier :)

(seems like there is no API to update the original value of a patched byte)